pkg/certify/external-bash/driver-call.go

```diff
@@ -0,0 +1,337 @@
+	tv, _ := volume.(*testVolume)
+	volHandle := b.DriverInfo.Name + "-vol"
+	fmt.Printf("\n\nVolume Handle: ------------------- %s\n\n", volHandle)
+	return &v1.PersistentVolumeSource{
+		CSI: &v1.CSIPersistentVolumeSource{
+			Driver:           b.DriverInfo.Name,
+			VolumeHandle:     volHandle,
+			VolumeAttributes: tv.volumeAttrib,
+		},
+	}, nil
+}
+
+func (b bashDriver) GetDynamicProvisionStorageClass(config *testsuites.PerTestConfig, fsType string) *storagev1.StorageClass {
+	f := config.Framework
+
+	if b.StorageClass.FromName {
+		provisioner := b.DriverInfo.Name
+		parameters := map[string]string{}
+		ns := f.Namespace.Name
+		suffix := provisioner + "-sc"
+		if fsType != "" {
+			parameters["csi.storage.k8s.io/fstype"] = fsType
+		}
+
+		return testsuites.GetStorageClass(provisioner, parameters, nil, ns, suffix)
+	}
+
+	items, err := f.LoadFromManifests(b.StorageClass.FromFile)
+	Expect(err).NotTo(HaveOccurred(), "load storage class from %s", b.StorageClass.FromFile)
+	Expect(len(items)).To(Equal(1), "exactly one item from %s", b.StorageClass.FromFile)
+
+	err = f.PatchItems(items...)
+	Expect(err).NotTo(HaveOccurred(), "patch items")
+
+	sc, ok := items[0].(*storagev1.StorageClass)
+	Expect(ok).To(BeTrue(), "storage class from %s", b.StorageClass.FromFile)
+	if fsType != "" {
+		if sc.Parameters == nil {
+			sc.Parameters = map[string]string{}
+		}
+		sc.Parameters["csi.storage.k8s.io/fstype"] = fsType
+	}
+	return sc
+}
+
+func (b bashDriver) GetSnapshotClass(config *testsuites.PerTestConfig) *unstructured.Unstructured {
+	if !b.SnapshotClass.FromName {
+		framework.Skipf("Driver %q does not support snapshotting - skipping", b.DriverInfo.Name)
+	}
+
+	snapshotter := config.GetUniqueDriverName()
+	parameters := map[string]string{}
+	ns := config.Framework.Namespace.Name
+	suffix := snapshotter + "-vsc"
+
+	return testsuites.GetSnapshotClass(snapshotter, parameters, ns, suffix)
+}
+
+func (b *bashDriver) GetClaimSize() string {
+	return b.ClaimSize
+}
+
+func (b *bashDriver) PrepareTest(f *framework.Framework) (*testsuites.PerTestConfig, func()) {
+	config := &testsuites.PerTestConfig{
+		Driver:         b,
+		Prefix:         "external",
+		Framework:      f,
+		ClientNodeName: b.ClientNodeName,
+	}
+	return config, func() {}
+}
+
+//Example call: getCommand("nfs", createVolume)
+func execCommand(pluginName string, cmdName string, currentNameSpace string) (error, []byte) {
+	namespaceErr := setCurrentNameSpace(currentNameSpace)
+	if namespaceErr != nil {
+		return namespaceErr, nil
+	}
+
+	fmt.Printf("Command = %s", ". ../../pkg/certify/external-bash/"+pluginName+" && "+cmdName)
+
+	cmd := exec.Command("bash", "-c", ". ../../pkg/certify/external-bash/"+pluginName+" && "+cmdName)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+
+	namespaceErr = setCurrentNameSpace("default")
```

Comment:
ideally we should set it to whatever it was before we changed it, not necessarily "default". I am not sure how, I guess you would need to parse `kubectl config current-context`

Reply:
I used `kubectl config view | grep namespace` for it.